File: ueueo-framework/ueueo-blob-storing/src/main/java/Abp/BlobStoring/BlobContainerConfiguration.cs
using System;
using System.Collections.Generic;
using JetBrains.Annotations;
using Volo.Abp.Collections;

namespace Volo.Abp.BlobStoring;

public class BlobContainerConfiguration
{
    /**
     * The provider to be used to store BLOBs of this container.
    */
    public Type ProviderType;// { get; set; }

    /**
     * Indicates whether this container is multi-tenant or not.
     *
     * If this is <code>false</code> and your application is multi-tenant,
     * then the container is shared by all tenants in the system.
     *
     * This can be <code>true</code> even if your application is not multi-tenant.
     *
     * Default: true.
    */
    public boolean IsMultiTenant;// { get; set; } = true;

    public ITypeList<IBlobNamingNormalizer> NamingNormalizers;//  { get; }

    @Nonnull private readonly Dictionary<String, Object> _properties;

    @Nullable private readonly BlobContainerConfiguration _fallbackConfiguration;

    public BlobContainerConfiguration(BlobContainerConfiguration fallbackConfiguration = null)
    {
        NamingNormalizers = new TypeList<IBlobNamingNormalizer>();
        _fallbackConfiguration = fallbackConfiguration;
        _properties = new Dictionary<String, Object>();
    }

    [CanBeNull]
    public T GetConfigurationOrDefault<T>(String name, T defaultValue = default)
    {
        return (T)GetConfigurationOrNull(name, defaultValue);
    }

    [CanBeNull]
    public Object GetConfigurationOrNull(String name, Object defaultValue = null)
    {
        return _properties.GetOrDefault(name) ??
               _fallbackConfiguration?.GetConfigurationOrNull(name, defaultValue) ??
               defaultValue;
    }

    [NotNull]
    public BlobContainerConfiguration SetConfiguration(@Nonnull String name, @Nullable Object value)
    {
        Check.NotNullOrWhiteSpace(name, nameof(name));
        Objects.requireNonNull(value, nameof(value));

        _properties[name] = value;

        return this;
    }

    [NotNull]
    public BlobContainerConfiguration ClearConfiguration(@Nonnull String name)
    {
        Check.NotNullOrWhiteSpace(name, nameof(name));

        _properties.Remove(name);

        return this;
    }
}

File: ueueo-framework/ueueo-blob-storing/src/main/java/com/ueueo/blobstoring/BlobContainerConfiguration.java
package com.ueueo.blobstoring;

import com.ueueo.exception.SystemException;
import org.springframework.lang.NonNull;
import org.springframework.lang.Nullable;
import org.springframework.util.Assert;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class BlobContainerConfiguration {
    /**
     * The provider to be used to store BLOBs of this container.
     */
    private Class<?> providerType;

    /**
     * Indicates whether this container is multi-tenant or not.
     *
     * If this is <code>false</code> and your application is multi-tenant,
     * then the container is shared by all tenants in the system.
     *
     * This can be <code>true</code> even if your application is not multi-tenant.
     *
     * Default: true.
     */
    private boolean isMultiTenant = true;

    private List<? extends IBlobNamingNormalizer> namingNormalizers;

    @NonNull
    private Map<String, Object> properties;

    @Nullable
    private BlobContainerConfiguration fallbackConfiguration;

    public BlobContainerConfiguration(BlobContainerConfiguration fallbackConfiguration) {
        this.namingNormalizers = new ArrayList<>();
        this.fallbackConfiguration = fallbackConfiguration;
        this.properties = new HashMap<>();
    }

    @Nullable
    public <T> T getConfigurationOrDefault(String name, T defaultValue) {
        return (T) getConfigurationOrNull(name, defaultValue);
    }

    @Nullable
    public Object getConfigurationOrNull(String name, Object defaultValue) {
        Object obj = properties.get(name);
        if (obj != null) {
            return obj;
        }
        if (fallbackConfiguration != null) {
            return fallbackConfiguration.getConfigurationOrNull(name, defaultValue);
        }
        return defaultValue;
    }

    public <T> T getConfiguration(@NonNull String name, Class<T> type) {
        return (T) getConfiguration(name);
    }

    public Object getConfiguration(@NonNull String name) {
        Object value = getConfigurationOrNull(name, null);
        if (value == null) {
            throw new SystemException(String.format("Could not find the configuration value for '%s'!", name));
        }

        return value;
    }

    @NonNull
    public BlobContainerConfiguration setConfiguration(@NonNull String name, @Nullable Object value) {
        Assert.hasLength(name, "name must not empty!");
        Assert.notNull(value, "value must not null!");

        properties.put(name, value);

        return this;
    }

    @NonNull
    public BlobContainerConfiguration clearConfiguration(@NonNull String name) {
        Assert.hasLength(name, "name must not empty!");

        properties.remove(name);

        return this;
    }

    public Class<?> getProviderType() {
        return providerType;
    }

    public void setProviderType(Class<?> providerType) {
        this.providerType = providerType;
    }

    public boolean isMultiTenant() {
        return isMultiTenant;
    }

    public void setMultiTenant(boolean multiTenant) {
        isMultiTenant = multiTenant;
    }

    public List<? extends IBlobNamingNormalizer> getNamingNormalizers() {
        return namingNormalizers;
    }
}
